integration/test_collection_model.py

```diff
@@ -41,7 +41,7 @@ def client():
     )
     client.collection_model.delete(Group)
     collection = client.collection_model.create(
-        CollectionModelConfig(model=Group, vectorizer=Vectorizer.NONE)
+        CollectionModelConfig[Group](model=Group, vectorizer=Vectorizer.NONE)
```

Comment:
This is the problem, which I don't think ever went away, of when and where types are applied in Python (but also in TypeScript):
- In order for the type of the model to be known at compile, it needs to be passed as a type argument.
- In order for the model itself to be known at runtime, so that it can be used to construct instances of the model as part of the ORM, it needs to be passed as a function argument.
- 
Without the `CollectionModelConfig[Group]`, the intellisense doesn't work correctly in the tests and I don't believe it ever did; it just slipped under our noses. This double specification can, however, be abstracted away from the user if we provide a `@classmethod` constructor. Something like:
```python
CollectionModelConfig.make(model=Group, vectorizer=Vectorizer.NONE)
```
where the `make()` function uses `model=Group` to construct:
```python
return CollectionModelConfig[config.model](model=config.model, vectorizer=config.vectorizer)
```
Do you think such an interface would be confusing for users at all?

Reply:
hmmmm - then let's leave it like it is for now